lib/protoparser/prometheus/parser.go

```diff
@@ -180,24 +237,25 @@ func (r *Row) unmarshal(s string, tagsPool []Tag, noEscapes bool) ([]Tag, error)
 }
 
 var rowsReadScrape = metrics.NewCounter(`vm_protoparser_rows_read_total{type="promscrape"}`)
+var metadataReadScrape = metrics.NewCounter(`vm_protoparser_metadata_read_total{type="promscrape"}`)
 
 func unmarshalRows(dst []Row, s string, tagsPool []Tag, noEscapes bool, errLogger func(s string)) ([]Row, []Tag) {
 	dstLen := len(dst)
 	for len(s) > 0 {
 		n := strings.IndexByte(s, '\n')
 		if n < 0 {
 			// The last line.
-			dst, tagsPool = unmarshalRow(dst, s, tagsPool, noEscapes, errLogger)
+			dst, tagsPool = unmarshalRow(s, dst, tagsPool, noEscapes, errLogger)
```

Comment:
It's a common parsing convention to have destination buffers as a first function args

Reply:
Fixed.